### PR TITLE
Initialize angel_msgs/angel_debug, Update py-nodes to use message

### PR DESCRIPTION
**Pull request by Purg**
This updates more than it ideally should but includes:
* docker/support updates
  * Include additional option to `angel-workspace-shell.sh` script to automatically source the built workspace setup script, making it easier to get a "use my workspace" environment up.
* Update smqtk-detection dep to use a pinned hash from master upstream. This should be temporary until there is a release with the desired fix in it. Pyproject/poetry.lock/Dockerfile have been updated to reflect/handle this.
* added angel_msgs, including initial detection message
* added a C++ package angel_debug with a very initial C++ version of the object detection debugging node. This is not complete but I didn't feel like extracting it from this history.
* Add an uncrustify config and helper script for standard C++ code style, useful for future use of CI enforcement.

### CODE REVIEW

File: angel-workspace-shell.sh [hunk 1]
```diff
@@ -4,24 +4,26 @@
...
 ./docker/docker-compose.yml configuration file.
 
-SERVICE         Name of the angel-system service to start a container for.
+Additional arguments, and explicitly those after a '--' are passed as command
+arguments to the service run.
 
 Options:
-  -h | --help   Display this message.
+  -h | --help                   Display this message.
+  -r | --run-setup              Start the service passing a
```
Comment (Purg) [inline angel-workspace-shell.sh:25]: Finish sentence

File: ros/angel_debug/package.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>angel_debug</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="root@todo.todo">root</maintainer>
+  <license>TODO: License declaration</license>
```
Comment (Purg) [inline ros/angel_debug/package.xml:8]: Fix temporary values

File: ros/angel_msgs/package.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>angel_msgs</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
+  <maintainer email="root@todo.todo">root</maintainer>
+  <license>TODO: License declaration</license>
```
Comment (Purg) [inline ros/angel_msgs/package.xml:8]: fixup temp values

File: ros/python/py_pubsub/py_pubsub/object_detector_debug.py [hunk 4]
```diff
@@ -117,10 +127,11 @@ def listener_callback(self, image):
...
-        plot.gcf().canvas.flush_events()
-        plot.show(block=False)
+        img_mat = np.asarray(pil_image)
+        img_msg = BRIDGE.cv2_to_imgmsg(img_mat, encoding="rgb8")
+        # img_msg = BRIDGE.cv2_to_compressed_imgmsg(img_mat, dst_format='jpg')
+        img_msg.header = image.header
+        self._pub_debug_detections_image.publish(img_msg)
 
 
 def convert_class_num_to_label(num):
```
Comment (joshanderson-kw) [inline ros/python/py_pubsub/py_pubsub/object_detector_debug.py:137]: Could remove this function now I think.

**Comment by joshanderson-kw** [comment]
`object_detector` and `object_detector_debug` work on my end. Did not spend too much time trying the angel_debug overlay node.

**Review by joshanderson-kw — APPROVED** [review]

**Pull request merged**